### PR TITLE
[all] Several STC fixes

### PR DESCRIPTION
**Pull request by damienmarchal**
Changelog:
-  Remove the plugins in deprecated directory 
-  Remove DETECTIONOUTPUT_BARYCENTRICINFO (https://github.com/sofa-framework/sofa/pull/1046)
 - Remove remaining SOFA_WITH_FLOAT (https://github.com/sofa-framework/sofa/pull/1047)
-  Replace 3.14 with M_PI (https://github.com/sofa-framework/sofa/pull/1045) 

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by guparan** [comment]
Why 4 PRs in one? The 4 PRs looked nice on their own.

**Comment by damienmarchal** [comment]
Because the three PR where build on top of each others...so they are pointless.

**Comment by damienmarchal** [comment]
[ci-build]

**Comment by epernod** [comment]
yes but 4 PR is easier to review...

File: applications/plugins/LeapMotion/LeapMotion/LeapMath.h [hunk 1]
```diff
@@ -17,7 +17,7 @@
 namespace Leap {
 
 /** The constant pi as a single precision floating point number. */
-static const float PI          = 3.1415926536f;
+static const float PI          = M_PI;
```
Comment (epernod) [inline applications/plugins/LeapMotion/LeapMotion/LeapMath.h]: why recreating a new variable?

File: applications/plugins/SofaDistanceGrid/components/collision/FFDDistanceGridDiscreteIntersection.cpp [hunk 2]
```diff
@@ -128,10 +128,6 @@ int FFDDistanceGridDiscreteIntersection::computeIntersection(FFDDistanceGridColl
                 {
                     detection->point[0] = grid1->meshPts[c1.points[i].index];
                     detection->point[1] = Vector3(p2) - grad * d;
-#ifdef DETECTIONOUTPUT_BARYCENTRICINFO
```
Comment (epernod) [inline applications/plugins/SofaDistanceGrid/components/collision/FFDDistanceGridDiscreteIntersection.cpp:131]: Do you know what was this macro used for?

File: modules/SofaMiscFem/TriangularFEMForceField.inl [hunk 3]
```diff
@@ -910,70 +910,6 @@ void TriangularFEMForceField<DataTypes>::computePrincipalStress(Index elementInd
 
     triangleInf[elementIndex].principalStressDirection = averageVector2;
 
-#ifdef PLOT_CURVE
```
Comment (epernod) [inline modules/SofaMiscFem/TriangularFEMForceField.inl:913]: and this one? not needed at all?

**Comment by damienmarchal** [comment]
:)

**Pull request merged**